devel/apache-arrow/Portfile

```diff
@@ -0,0 +1,223 @@
+
+long_description    Apache Arrow is a development platform for in-memory \
+                    analytics. It contains a set of technologies that \
+                    enable big data systems to process and move data \
+                    fast.
+
+checksums           rmd160  26a1438b995ca7c3fc8ff65d703bfb29133f5ac1 \
+                    sha256  56cc21a24d68a24ccdff043b4a4cec57206a3be5e5b128608c81dbc20dd843ce \
+                    size    6847581
+
+compiler.cxx_standard   2011
+
+universal_variant   no
+
+# Build instructions: https://github.com/apache/arrow/blob/master/docs/source/developers/python.rst#build-and-test
+cmake.source_dir    ${worksrcpath}/cpp
+cmake.build_type    Release
+
+set llvm_version    10
+set llvm_prefix     ${prefix}/libexec/llvm-${llvm_version}
+
+cmake.module_path-append \
+                    ${llvm_prefix}/lib/cmake/llvm
+
+depends_build-append \
+                    port:pkgconfig
+
+depends_lib-append \
+                    port:llvm-${llvm_version}
+
+pre-configure {
+    xinstall -d ${workpath}/bin
+    foreach f [glob -nocomplain ${prefix}/bin/ll*-${llvm_version}] {
+        ln -s ${f} ${workpath}/bin/[strsed [file tail ${f}] "s|-${llvm_version}$||"]
+    }
+}
+
+configure.args-append \
+                    -DARROW_DEPENDENCY_USE_SHARED:BOOL=ON \
+                    -DARROW_FLIGHT:BOOL=ON \
+                    -DARROW_JEMALLOC:BOOL=ON \
+                    -DARROW_ORC:BOOL=ON \
+                    -DARROW_PARQUET:BOOL=ON \
+                    -DARROW_PLASMA:BOOL=ON \
+                    -DARROW_PROTOBUF_USE_SHARED=:BOOLON \
+                    -DARROW_PYTHON:BOOL=ON \
+                    -DARROW_USE_CCACHE:BOOL=OFF \
+                    -DARROW_WITH_BZ2:BOOL=ON \
+                    -DARROW_WITH_ZLIB:BOOL=ON \
+                    -DARROW_WITH_ZSTD:BOOL=ON \
+                    -DARROW_WITH_LZ4:BOOL=ON \
+                    -DARROW_WITH_SNAPPY:BOOL=ON \
+                    -DARROW_WITH_BROTLI:BOOL=ON \
+                    -DARROW_INSTALL_NAME_RPATH:BOOL=OFF \
+                    -DgRPC_ROOT=${prefix} \
+                    -DLLVM_ROOT=${llvm_prefix}
+
+# used for both apache-arrow and py*-pyarrow
+destroot.target     install
+
+if {${name} eq ${subport}} {
+    # Apache Arrow requires a python3 version to build; use this
+    set python_branch   3.8
+    set python_version  [join [split ${python_branch} .] ""]
+    set python_prefix   ${frameworks_dir}/Python.framework/Versions/${python_branch}
+    set python_bin      ${python_prefix}/bin/python${python_branch}
+
+    depends_build-append \
+                    port:python${python_version} \
+                    port:py${python_version}-cython \
+                    port:py${python_version}-numpy
+
+    depends_lib-append \
+                    port:boost \
+                    port:brotli \
+                    port:bzip2 \
+                    port:c-ares \
+                    port:flatbuffers \
```

Comment:
The developers of flatbuffers do not believe that it should be part of a package manager like MacPorts; rather, it should be bundled with whatever software wants to use it. See https://github.com/google/flatbuffers/issues/6076#issuecomment-694954041. If and when we remove flatbuffers from MacPorts, it would break this port. Does apache-arrow have a bundled copy of flatbuffers that it could use instead? If not, could you suggest it to them? Also FYI @herbygillot

Reply:
Upstream has it: https://github.com/apache/arrow/tree/master/cpp/thirdparty

I verify that `port test` works without MacPorts `flatbuffers`.